src/api/app/components/notification_action_bar_component.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class NotificationActionBarComponent < ApplicationComponent
+  attr_accessor :type, :update_path
+
+  def initialize(type:, update_path:)
+    super
+
+    @type = type
+    @update_path = update_path + '&update_all=true'
+  end
+
+  def button_text(all: false)
+    if all
+      type == 'read' ? "Mark all as 'Unread'" : "Mark all as 'Read'"
+    else
+      type == 'read' ? "Mark as 'Unread'" : "Mark as 'Read'"
+    end
```

Comment:
I find this combination of conditions and ternaries difficult to read. Maybe rewriting it as guard clauses it's easier to read?

Reply:
Thanks for the suggestion. It's updated :slightly_smiling_face: